### PR TITLE
Swagger-codegen and swagger-parser orbits[for APIM]

### PR DESCRIPTION
**Pull request by rnavagamuwa**
We are currently implementing a new feature for APIM and with this feature, client side sdk for subscribed APIs can be generated automatically. Since we are using Swagger-Codegen for the implementation it is necessary that we orbit these two libraries.
JIRA key :- **APIMANAGER-4598**

### CODE REVIEW

File: swagger-codegen/2.1.5.wso2v1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        This bundle will export packages from swagger-codegen libraries of io.swagger
+    </description>
+    <url>http://wso2.org</url>
+
+
+    <dependencies>
+        <dependency>
+            <groupId>io.swagger</groupId>
+            <artifactId>swagger-codegen</artifactId>
+            <version>2.1.5</version>
```
Comment (thusithathilina) [inline swagger-codegen/2.1.5.wso2v1/pom.xml]: make all dependencies optional

File: swagger-parser/1.0.17.wso2v1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        <dependency>
+            <groupId>io.swagger</groupId>
+            <artifactId>swagger-parser</artifactId>
+            <version>1.0.17</version>
+        </dependency>
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>1.4</version>
+        </dependency>
```
Comment (thusithathilina) [inline swagger-parser/1.0.17.wso2v1/pom.xml]: dependencies should be optional

File: swagger-codegen/2.1.5.wso2v1/pom.xml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                        <Private-Package>
+                        </Private-Package>
+                        <Include-Resource>
+                            {maven-resources}
+                        </Include-Resource>
+                        <Embed-Dependency>
+                            joda-time;scope=compile|runtime;inline=false,
+                            commons-io;scope=compile|runtime;inline=false
+                        </Embed-Dependency>
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
Comment (thusithathilina) [inline swagger-codegen/2.1.5.wso2v1/pom.xml]: Can you verify the usages of DynamicImports?

Comment (rnavagamuwa) [inline swagger-codegen/2.1.5.wso2v1/pom.xml]: This library dynamically loades some classes using java.lang.Class.forName() method and that's whyI have used DynamicaImports.

**Pull request merged**